applications/asset_tracker_v2/tests/lwm2m_integration/src/lwm2m_integration_test.c

```diff
@@ -327,86 +328,59 @@ void test_lwm2m_integration_network_error(void)
 	TEST_ASSERT_EQUAL(CLOUD_WRAP_EVT_ERROR, last_cb_type);
 }
 
-void test_lwm2m_integration_fota_result_get(void)
-{
-	/* Expect the FOTA update result to be retrieved for any update in FOTA state. */
-	__cmock_lwm2m_get_u8_ExpectAndReturn(&LWM2M_OBJ(5, 0, 5), NULL, 0);
-	__cmock_lwm2m_get_u8_IgnoreArg_value();
-
-	__cmock_lwm2m_get_u8_ExpectAndReturn(&LWM2M_OBJ(5, 0, 5), NULL, 0);
-	__cmock_lwm2m_get_u8_IgnoreArg_value();
-
-	__cmock_lwm2m_get_u8_ExpectAndReturn(&LWM2M_OBJ(5, 0, 5), NULL, 0);
-	__cmock_lwm2m_get_u8_IgnoreArg_value();
-
-	__cmock_lwm2m_get_u8_ExpectAndReturn(&LWM2M_OBJ(5, 0, 5), NULL, 0);
-	__cmock_lwm2m_get_u8_IgnoreArg_value();
-	__cmock_lwm2m_firmware_set_update_state_cb_Expect(NULL);
-
-	firmware_update_state_cb(STATE_IDLE);
-	firmware_update_state_cb(STATE_DOWNLOADING);
-	firmware_update_state_cb(STATE_DOWNLOADED);
-	firmware_update_state_cb(STATE_UPDATING);
-}
-
-void test_lwm2m_integration_fota_result_get_error(void)
-{
-	__cmock_lwm2m_get_u8_ExpectAnyArgsAndReturn(-1);
-
-	firmware_update_state_cb(STATE_DOWNLOADING);
-	TEST_ASSERT_EQUAL(CLOUD_WRAP_EVT_ERROR, last_cb_type);
-}
-
 void test_lwm2m_integration_fota_error(void)
 {
-	__cmock_lwm2m_get_u8_IgnoreAndReturn(0);
+	struct lwm2m_fota_event event = {
+		.id = LWM2M_FOTA_UPDATE_ERROR,
+		.failure.obj_inst_id = 0,
+		.failure.update_failure = 10
+	};
 
 	/* Expect an error event to be returned if FOTA state reverts to STATE_IDLE. */
-	firmware_update_state_cb(STATE_IDLE);
+	firmware_update_events_cb(&event);
 	TEST_ASSERT_EQUAL(CLOUD_WRAP_EVT_FOTA_ERROR, last_cb_type);
 }
 
 void test_lwm2m_integration_fota_downloading(void)
 {
-	__cmock_lwm2m_get_u8_IgnoreAndReturn(0);
+	struct lwm2m_fota_event event = {
+		.id = LWM2M_FOTA_DOWNLOAD_START,
+		.download_start.obj_inst_id = 0
+	};
 
-	firmware_update_state_cb(STATE_DOWNLOADING);
+	firmware_update_events_cb(&event);
 	TEST_ASSERT_EQUAL(CLOUD_WRAP_EVT_FOTA_START, last_cb_type);
 }
 
 void test_lwm2m_integration_fota_downloaded(void)
 {
-	__cmock_lwm2m_get_u8_IgnoreAndReturn(0);
+	struct lwm2m_fota_event event = {
+		.id = LWM2M_FOTA_DOWNLOAD_FINISHED,
+		.download_ready.obj_inst_id = 0
+	};
 
 	/* Expect no event to be called by setting last_cb_type to UINT8_MAX and verifying that
 	 * the value has not changed after the state change.
 	 */
 	last_cb_type = UINT8_MAX;
 
-	firmware_update_state_cb(STATE_DOWNLOADED);
+	firmware_update_events_cb(&event);
 	TEST_ASSERT_EQUAL(UINT8_MAX, last_cb_type);
 }
 
 void test_lwm2m_integration_fota_updating(void)
```

Comment:
May be change the names of these tests to reflect what kind of events they are sending to the callback?

For example this one can be

```suggestion
void test_lwm2m_integration_fota_update_image_req(void)
```